rclc/src/rclc/executor.c

```diff
@@ -906,6 +1143,177 @@ _rclc_take_new_data(rclc_executor_handle_t * handle, rcl_wait_set_t * wait_set)
+        if (rc != RCL_RET_OK) {
+          PRINT_RCLC_ERROR(rclc_take_new_data, rcl_action_take_goal_response);
+          RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Error number: %d", rc);
+          return rc;
+        }
+        rclc_action_goal_handle_t * goal_handle =
+          rclc_action_get_handle_by_goal_request_sequence_number(
+          handle->action_client, aux_goal_response_header.sequence_number);
+        if (NULL != goal_handle) {
+          goal_handle->available_goal_response = true;
+          goal_handle->goal_accepted = aux_goal_response.accepted;
+        }
+      }
+      if (handle->action_client->feedback_available) {
+        rc = rcl_action_take_feedback(
+          &handle->action_client->rcl_handle,
+          handle->action_client->ros_feedback);
+
+        if (rc != RCL_RET_OK) {
+          PRINT_RCLC_ERROR(rclc_take_new_data, rcl_action_take_feedback);
+          RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Error number: %d", rc);
+          return rc;
+        }
+
+        rclc_action_goal_handle_t * goal_handle = rclc_action_get_handle_by_uuid(
+          handle->action_client,
+          &handle->action_client->ros_feedback->goal_id);
+        if (NULL != goal_handle) {
+          goal_handle->available_feedback = true;
+        }
+      }
+      if (handle->action_client->cancel_response_available) {
+        rmw_request_id_t cancel_response_header;
+        rc = rcl_action_take_cancel_response(
+          &handle->action_client->rcl_handle,
+          &cancel_response_header,
+          &handle->action_client->ros_cancel_response);
+
+        if (rc != RCL_RET_OK) {
+          PRINT_RCLC_ERROR(rclc_take_new_data, rcl_action_take_result_response);
+          RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Error number: %d", rc);
+          return rc;
+        }
+
+        rclc_action_goal_handle_t * goal_handle =
+          rclc_action_get_handle_by_cancel_request_sequence_number(
+          handle->action_client,
+          cancel_response_header.sequence_number);
+
+        if (NULL != goal_handle) {
+          goal_handle->available_cancel_response = true;
+          goal_handle->goal_cancelled = false;
+          for (size_t i = 0; i < handle->action_client->ros_cancel_response.goals_canceling.size;
+            i++)
+          {
+            rclc_action_goal_handle_t * aux = rclc_action_get_handle_by_uuid(
+              handle->action_client,
+              &handle->action_client->ros_cancel_response.goals_canceling.data[i].goal_id);
+            if (NULL != aux) {
+              goal_handle->goal_cancelled = true;
+            }
+          }
+        }
+      }
+      if (handle->action_client->result_response_available) {
+        rmw_request_id_t result_request_header;
+        rc = rcl_action_take_result_response(
+          &handle->action_client->rcl_handle,
+          &result_request_header,
+          handle->action_client->ros_result_response);
+
+        if (rc != RCL_RET_OK) {
```

Comment:
```suggestion
          handle->action_client->ros_result_response);
        if (rc != RCL_RET_OK) {
```
For consistency with above code layout.